lib/elixir/src/elixir_erl.erl

```diff
@@ -138,7 +138,13 @@ elixir_to_erl_cons2([], Acc) ->
 scope(_Meta) ->
   #elixir_erl{}.
 
-%% Compilation hook.
+%% Static compilation hook, used in protocol consolidation
+
+compile(Map, TypeSpecs, Chunks) ->
```

Comment:
Let's call this `consolidate` for now, it is very specific to the protocol mechanism and not a general compilation mechanism.

Reply:
I was thinking about moving the ETS table access up to the callers of `compile/1` and have a pure function here in general.

:+1: for `consolidate/3`.